### PR TITLE
r/wafv2_web_acl_logging_configuration: update redacted_fields schema definition

### LINKED ISSUE #14244: redacted_fields "single_query_argument" for resource aws_wafv2_web_acl_logging_configuration throws an error
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform AWS Provider Version

* Terraform version: 0.13.0
* AWS Provider Version: 2.70.0

### Affected Resource(s)

* aws_wafv2_web_acl_logging_configuration

### Terraform Configuration Files

```hcl
resource "aws_wafv2_web_acl_logging_configuration" "example" {
  log_destination_configs = ["${aws_kinesis_firehose_delivery_stream.example.arn}"]
  resource_arn            = "${aws_wafv2_web_acl.example.arn}"
  redacted_fields {
    single_query_argument {
      name = "username"
    }
  }
}
```
### Expected Behavior
The data field for Query string is selected on the AWS console as shown below:
image

### Actual Behavior
The API call breaks, throwing the error shown below:

image

### Steps to Reproduce

1. `terraform apply`

### LINKED ISSUE #14248: redacted_fields arguments for resource aws_wafv2_web_acl_logging_configuration are not making changes on AWS
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform AWS Provider Version

- Terraform version: 0.13.0

- AWS Provider Version: 2.70.0

### Affected Resource(s)

* aws_wafv2_web_acl_logging_configuration

### Terraform Configuration Files

```hcl
resource "aws_wafv2_web_acl_logging_configuration" "example" {
  log_destination_configs = ["${aws_kinesis_firehose_delivery_stream.example.arn}"]
  resource_arn            = "${aws_wafv2_web_acl.example.arn}"
  redacted_fields {
    all_query_arguments { }
    uri_path { }
    method { }
    body { }
    query_string { }
  }
}
```
### Expected Behavior

The data fields passed are selected on the AWS console as shown below:
image
### Actual Behavior

Some fields are not present on the AWS Console and even the ones present do not seem to take effect - there's no error message - but also there's no effect.
Also, the configurations are are identified as new changes each time "terraform plan" is run 
image

### Steps to Reproduce

1. `terraform apply`

### PR DESCRIPTION
**Pull request by anGie44**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #14248 
Closes #14244 

### Notes: 
The linked GH-Issues brought to light the following:
* `redacted_fields` blocks in the context of a `LoggingConfiguration` only permits a subset of the fields documented in https://docs.aws.amazon.com/waf/latest/APIReference/API_FieldToMatch.html and the API will error when an unsupported field (e.g. single_query_argument) is provided --> to account for this behavior, a custom schema is made for this argument instead of using the shared one in `waf_helper.go`; fields that are not supported have been marked as `Deprecated` until they can officially be removed

* only 1 nested attribute is valid in the `redacted_fields` configuration block of `aws_wafv2_web_acl_logging_configuration` and the `field_to_match` block in the `wafv2_web_acl` and `wafv2_rule_group` resources --> to account for this, the documentation has been updated to prevent users from setting multiple attributes

* the previous `Set` type used for `redacted_fields` would be evaluated as an empty object when fields with empty schemas (e.g. `body`) were configured --> to address this, the field argument has been changed to type `List`

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
resource/wafv2_logging_configuration: update redacted_fields handling
```

Output from acceptance testing:

```
🏃‍♀️ 🏃‍♂️ 🏃‍♀️ 🏃‍♂️ sooo fast, terraform 0.14 💪  
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_disappears (88.40s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_webACLDisappears (97.98s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_updateQueryStringRedactedField (102.55s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_updateEmptyRedactedFields (118.02s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_updateUriPathRedactedField (123.19s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_emptyRedactedFields (124.89s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_basic (132.33s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_updateMethodRedactedField (140.09s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_updateMultipleRedactedFields (147.57s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_updateSingleHeaderRedactedField (154.22s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_changeResourceARNForceNew (199.44s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_changeLogDestinationConfigsForceNew (200.19s)
```

### CODE REVIEW

File: website/docs/r/wafv2_web_acl_logging_configuration.html.markdown
Comment (dennis-grunert) [inline website/docs/r/wafv2_web_acl_logging_configuration.html.markdown:49]: The word _inspect_ seems wrong here. It was probably copied from the `field_to_match` in the `aws_wafv2_web_acl` resource. But here we are not matching a field but redacting it. My suggestion would be to replace _Inspect_ by _Redact_.

The same applies to the _Single Header_ section.

Comment (anGie44) [reply website/docs/r/wafv2_web_acl_logging_configuration.html.markdown:49]: Agreed, thanks for commenting here! The definition of these params are definitely context-specific. I think they should all probably be updated to use `redact` or something along the lines of `hide from logs` . will update!

Comment (anGie44) [reply website/docs/r/wafv2_web_acl_logging_configuration.html.markdown]: changes added in c7429ef :)

**Comment by bflad** [comment]
@anGie44 can you please rebase this pull request?

File: aws/resource_aws_wafv2_web_acl_logging_configuration.go
Comment (anGie44) [inline aws/resource_aws_wafv2_web_acl_logging_configuration.go:53]: maybe we can skip this deprecation part (since they don't function as is?)

File: aws/resource_aws_wafv2_web_acl_logging_configuration.go
Comment (anGie44) [inline aws/resource_aws_wafv2_web_acl_logging_configuration.go:47]: understandably not an ideal side-effect  :/ any thoughts on using a different hack to work around the order diffs?

File: aws/resource_aws_wafv2_web_acl_logging_configuration.go
Comment (gdavison) [reply aws/resource_aws_wafv2_web_acl_logging_configuration.go:54]: Yeah, if it's never worked, I think we're OK to just remove it

**Review by gdavison — APPROVED** [review]
Looks good to me 🚀

Waiting on the acceptance tests

**Review by gdavison — APPROVED** [review]
Acceptance tests

--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_updateUriPathRedactedField (99.34s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_disappears (116.45s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_emptyRedactedFields (116.61s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_basic (120.68s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_updateMultipleRedactedFields (125.83s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_disappears_WebAcl (129.62s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_updateQueryStringRedactedField (135.83s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_updateEmptyRedactedFields (138.26s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_updateSingleHeaderRedactedField (149.56s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_updateMethodRedactedField (150.68s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_changeLogDestinationConfigsForceNew (182.36s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_changeResourceARNForceNew (200.66s)

**Comment by denis-khalturin-incountry** [comment]
@gdavison, are we awaiting a second review for a PR merge?

**Comment by anGie44** [comment]
Hi @denis-khalturin-incountry , this one slipped my radar! Waiting on CI checks and then will merge 👍

**Comment by anGie44** [comment]
Output of acceptance tests (commercial):
```
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_disappears (104.02s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_basic (111.54s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_updateMethodRedactedField (138.00s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_updateMultipleRedactedFields (144.81s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_updateEmptyRedactedFields (147.75s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_updateQueryStringRedactedField (150.65s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_disappears_WebAcl (152.47s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_emptyRedactedFields (153.96s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_updateUriPathRedactedField (162.43s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_updateSingleHeaderRedactedField (179.40s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_changeLogDestinationConfigsForceNew (198.54s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_changeResourceARNForceNew (238.30s)
```

gov cloud:
```
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_emptyRedactedFields (70.81s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_disappears_WebAcl (73.56s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_updateQueryStringRedactedField (99.73s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_disappears (100.06s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_basic (107.22s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_updateEmptyRedactedFields (107.70s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_updateUriPathRedactedField (109.45s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_updateMethodRedactedField (121.89s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_updateSingleHeaderRedactedField (124.12s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_updateMultipleRedactedFields (124.39s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_changeLogDestinationConfigsForceNew (146.54s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_changeResourceARNForceNew (180.79s)
```

**Pull request merged**
